src/openms/source/ANALYSIS/ID/IDBoostGraph.cpp

```diff
@@ -954,7 +978,50 @@ namespace OpenMS
   }
   */
 
-  void IDBoostGraph::resolveGraphPeptideCentric_(Graph& fg/*, bool resolveTies*/)
+  void IDBoostGraph::resolveGraphPeptideCentric(bool removeAssociationsInData/*, bool resolveTies*/)
+  {
+    if (ccs_.empty() && boost::num_vertices(g) == 0)
+    {
+      throw Exception::MissingInformation(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION, "Graph empty. Build it first.");
+    }
+
+    ProgressLogger pl;
+    pl.setLogType(ProgressLogger::CMD);
+
+    if (ccs_.empty())
+    {
+      pl.startProgress(0, 1, "Resolving graph...");
+      resolveGraphPeptideCentric_(g, removeAssociationsInData);
+      pl.nextProgress();
+      pl.endProgress();
+    }
+    else
+    {
+      pl.startProgress(0, ccs_.size(), "Resolving graph...");
+      #pragma omp parallel for
+      for (int i = 0; i < static_cast<int>(ccs_.size()); i += 1)
+      {
+        Graph& curr_cc = ccs_.at(i);
+
+        #ifdef INFERENCE_MT_DEBUG
+        OPENMS_LOG_INFO << "Processing on thread# " << omp_get_thread_num() << std::endl;
+        #endif
+
+        #ifdef INFERENCE_DEBUG
+        OPENMS_LOG_INFO << "Processing cc " << i << " with " << boost::num_vertices(curr_cc) << " vertices." << std::endl;
+        OPENMS_LOG_INFO << "Printing cc " << i << std::endl;
+        printGraph(LOG_INFO, curr_cc);
+        OPENMS_LOG_INFO << "Printed cc " << i << std::endl;
+        #endif
+
+        resolveGraphPeptideCentric_(curr_cc, removeAssociationsInData);
+        pl.setProgress(i);
```

Comment:
this will make the progress jump forth and back...
Depending on how expensive `resolveGraphPeptideCentric_` is, you can use 
`#pragma omp parallel for schedule(dynamic, 1)` to only process 1 chunk per thread at a time and then only use
`#pragma omp master` to set the progress. This will be somewhat accurate then.
For full accuracy, you'd need an atomic counter and report progress in masterthread.